### ISSUE TITLE
`FORCE_COLOR=true` causes "Cache folder path is retrieved for yarn but doesn't exist on disk"

### ISSUE THREAD

**Issue by bencochran** [actions/setup-node#772]
## Description:
**tl;dr:** `FORCE_COLOR=true` in build environment causes `yarn cache dir` output to be prefixed with "`\u001b[2K\u001b[1G`" which gets pulled in as  [`cachePath`](https://github.com/actions/setup-node/blob/eaead3517c1eac8d9276d5964db5d7572d13a396/src/cache-save.ts#L34), which definitely doesn’t [`fs.existsSync`](https://github.com/actions/setup-node/blob/eaead3517c1eac8d9276d5964db5d7572d13a396/src/cache-save.ts#L39).

**Workaround at the end** if google leads you here before this is fixed.

The story version:
I kept having failures on [“Post Run actions/setup-node@v3”](https://github.com/bencochran/DeePhone/actions/runs/5165272136/jobs/9304807664) on a [seemingly minimal yarn action](https://github.com/bencochran/DeePhone/blob/main/.github/workflows/test.yml):

```yaml
on: [push, pull_request]

env:
  FORCE_COLOR: true

jobs:
  test:
    name: test
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-node@v3
      with:
        node-version: 18
        cache: 'yarn'
    - run: yarn install --frozen-lockfile
    - run: yarn test
```

The failure I was getting:

```
Error: Cache folder path is retrieved for yarn but doesn't exist on disk: /home/runner/.cache/yarn/v6
```

I connected to the worker to poke around… that directory was indeed there. I try `fs.existSync`ing the directory myself from node, thinking it’s something wacky with permissions… works fine.

Well… it turns out the [`cachePath`](https://github.com/actions/setup-node/blob/eaead3517c1eac8d9276d5964db5d7572d13a396/src/cache-save.ts#L34) is `\u001b[2K\u001b[1G/home/runner/.cache/yarn/v6` with some pesky ANSI control characters hanging around. Copy-pasting it works just fine but, of course, those characters are in the js string.

`yarn cache dir` itself seems to throw those in to erase the entire line and move the cursor to the first column. Thanks, yarn.

### Action version:
3.4.1

### Platform:
- [X] Ubuntu
- [ ] macOS
- [ ] Windows

### Runner type:
- [X] Hosted
- [ ] Self-hosted

### Tools version:
- node: 18.16.0
- yarn: 1.22.19

### Repro steps:
Minimal repro repo: https://github.com/bencochran/setup-node-assi-escape-code-bug

Or:
1. Enable yarn caching
```
- uses: actions/setup-node@v3
  with:
    node-version: current
    cache: 'yarn'
```

2. Add `FORCE_COLOR=true`

```
env:
  FORCE_COLOR: true
```

### Expected behavior:
I expect the cache to work. And not fail the job.

### Actual behavior:
The cache doesn't work and the job fails at the post-step action with the following:
```
Post Run actions/setup-node@v3” 

Post job cleanup.
/usr/local/bin/yarn --version
1.22.19
/usr/local/bin/yarn cache dir
/home/runner/.cache/yarn/v6
Error: Cache folder path is retrieved for yarn but doesn't exist on disk: /home/runner/.cache/yarn/v6
```

## Workaround:
Remove `FORCE_COLOR` or move the `env` into a specific step like:

```
on: [push, pull_request]

jobs:
  test:
    name: test
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-node@v3
      with:
        node-version: current
        cache: 'yarn'
    - run: yarn install --frozen-lockfile
      env:
        FORCE_COLOR: true
    - run: yarn test
```

**Comment by dmitry-shibanov** [comment]
Hello @bencochran. Could you please try to use the snippet from this comment: https://github.com/actions/setup-node/issues/317#issuecomment-1065124272

**Comment by bencochran** [comment]
Thanks pointing me to that, @dmitry-shibanov. That [also make it work](https://github.com/bencochran/setup-node-assi-escape-code-bug/actions/runs/5179085355). Embarrassingly, I did see that issue, but didn’t read deep enough into the comments to find that it came down to the same issue. Sorry about the dupe!

**Comment by dmitry-shibanov** [comment]
Hello @bencochran. Sorry for the late response. For now I'm going to close the issue because it can be fixed by `FORCE_COLOR: 0` for action step. Besides, I think it can be fixed in scope of the action because it gets applied to the command output.
